homeassistant/components/plaato/__init__.py

```diff
@@ -59,32 +82,136 @@
     extra=vol.ALLOW_EXTRA,
 )
 
+SCAN_INTERVAL = timedelta(minutes=10)
+
 
-async def async_setup(hass, hass_config):
+async def async_setup(hass: HomeAssistant, config: dict):
     """Set up the Plaato component."""
+    hass.data.setdefault(DOMAIN, {})
     return True
 
 
-async def async_setup_entry(hass, entry):
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Configure based on config entry."""
-    if DOMAIN not in hass.data:
-        hass.data[DOMAIN] = {}
 
-    webhook_id = entry.data[CONF_WEBHOOK_ID]
-    hass.components.webhook.async_register(DOMAIN, "Plaato", webhook_id, handle_webhook)
+    use_webhook = entry.data.get(CONF_USE_WEBHOOK, False)
 
-    hass.async_create_task(hass.config_entries.async_forward_entry_setup(entry, SENSOR))
+    if use_webhook:
+        setup_webhook(hass, entry)
+    else:
+        await async_setup_coordinator(hass, entry)
+
+    for platform in PLATFORMS:
+        if entry.options.get(platform, True):
+            hass.async_create_task(
+                hass.config_entries.async_forward_entry_setup(entry, platform)
+            )
 
     return True
 
 
-async def async_unload_entry(hass, entry):
+def setup_webhook(hass: HomeAssistant, entry: ConfigEntry):
+    """Init webhook based on config entry."""
+    if entry.data[CONF_WEBHOOK_ID] is not None:
+        webhook_id = entry.data.get(CONF_WEBHOOK_ID)
+        device_name = entry.data.get(CONF_DEVICE_NAME)
+        device_type = entry.data.get(CONF_DEVICE_TYPE)
+        device = {DEVICE_NAME: device_name, DEVICE_TYPE: device_type}
+        undo_listener = entry.add_update_listener(_async_update_listener)
+
+        hass.data[DOMAIN][entry.entry_id] = {
+            COORDINATOR: None,
+            DEVICE: device,
+            SENSOR_DATA: None,
+            UNDO_UPDATE_LISTENER: undo_listener,
+        }
+
+        hass.components.webhook.async_register(
+            DOMAIN, f"{DOMAIN}.{device_name}", webhook_id, handle_webhook
+        )
+    else:
+        raise InvalidStateError
```

Comment:
```suggestion
    if entry.data[CONF_WEBHOOK_ID] is None:
        raise InvalidStateError

    webhook_id = entry.data[CONF_WEBHOOK_ID]
    device_name = entry.data[CONF_DEVICE_NAME]
    device_type = entry.data[CONF_DEVICE_TYPE]
    device = {DEVICE_NAME: device_name, DEVICE_TYPE: device_type}
    undo_listener = entry.add_update_listener(_async_update_listener)

    hass.data[DOMAIN][entry.entry_id] = {
        COORDINATOR: None,
        DEVICE: device,
        SENSOR_DATA: None,
        UNDO_UPDATE_LISTENER: undo_listener,
    }

    hass.components.webhook.async_register(
        DOMAIN, f"{DOMAIN}.{device_name}", webhook_id, handle_webhook
    )
```

Reply:
Raise right away so you can outdent the block below.

Please use `entry.data[XXXX]` instead of `entry.data.get(XXXX)` for required fields